src/operator/numpy/np_indexing_op.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Copyright (c) 2017 by Contributors
+ * \file np_indexing_op.h
+ * \brief Function definition of numpy indexing operator
+*/
+#ifndef MXNET_OPERATOR_NUMPY_NP_INDEXING_OP_H_
+#define MXNET_OPERATOR_NUMPY_NP_INDEXING_OP_H_
+
+#include <vector>
+#include "../contrib/boolean_mask-inl.h"
+#include "../tensor/indexing_op.h"
+#include "../tensor/broadcast_reduce_op.h"
+#ifdef __CUDACC__
+#include "../tensor/indexing_op-inl.cuh"
+#endif
+
+namespace mxnet {
+namespace op {
+
+namespace np_indexing_ {  // to avoid name conflict
+enum Inputs {kArr, kIdx};
+enum Outputs {kOut};
+}  // namespace np_indexing_
```

Comment:
Will we reuse these enums? Can we put them to `.cc`?

Reply:
We need use them in `.cu` file.